### DISCUSSION TITLE
TFT_eSPI and the not supported DMA on the 3.5" ili9488 SPI display

### DISCUSSION THREAD

**Discussion by Sentinel8000** [Bodmer/TFT_eSPI#2153] [New processor/display support or capability request]
There is any chanse in the future support DMA for  3.5" ili9488 SPI display?

I have tryed my ili9488 with lovyan GFX and looks wont have problem with DMA handling.

However the code working well, i want try out what a speed can i get with this library. I dont like work with the japan language commented lovyan library.

**Comment by Bodmer** [comment]
I don't intend to add DMA for 18 bit displays. This is mainly because the library fundamentally works with 16 bit colours at the moment and conversion to 18 bit caries a significant processor overhead which negates the benefits of DMA. There are plenty of TFT controllers that support 16 bit colours in SPI mode so it is not a priority.

At the moment my favorite processor is the RP2040 which may be able to use the PIO hardware to do colour conversion during DMA but lack of time prevents development of new features at the moment.

**Reply by Sentinel8000** [reply]
Thank you for your answer. 

Locally i have no large options selecting displays if i want use larger, 3.5" size display. 

Im working with a esp32-cam, so however i have a parallel 3.5" ili9486 display, i can use only with a normal Esp32. With a esp32 the parallel display working well with TFT_eSPI.

So, lookslike im stuck in the lovyan gfx driver. My main problem with this, your driver is much more user friendly and well documented against this other driver. Also i have with the library developer bad experience, i want ask 1-2 things, but the guy have trouble answering any questions.

I hope later i can get a compatible, larger size SPI display and i can try out your driver under working.

Have a nice day and also thanks for your work

**Reply by THLN33** [reply]
Hello,

I also use a Raspberry Pi Pico card (RP2040) with an LCD screen managed by the ILI9488 circuit. DMA transfers do not work with the original TFT_eSPI library. The image is displayed with wrong colors.

I made a minimal program to convert RGB 5-6-5 into RGB 6-6-6 colors using PIO, which makes it possible to use DMA for fast transfers. 

see the code here : https://github.com/THLN33/RP_PIO_SPI_16_24

This works fine, so it remains to integrate it into the TFT_eSPI library if it is useful to you.

Happy tinkering, See you soon!
